File: FruitVerse Snake/ScreenUtil.java
import javafx.geometry.Rectangle2D;
import javafx.stage.Screen;

public class ScreenUtil {
    private static double width = -1, height = -1;
    public static double getWidth() {
        if (width < 0) init();
        return width;
    }
    public static double getHeight() {
        if (height < 0) init();
        return height;
    }
    private static void init() {
        Rectangle2D b = Screen.getPrimary().getVisualBounds();
        width = b.getWidth();
        height = b.getHeight();
    }
}

File: js/ScreenUtil.js
// ============================================
// ScreenUtil — Responsive screen sizing utility
// Equivalent to Java ScreenUtil class
// ============================================
export class ScreenUtil {
    constructor() {
        this.width = window.innerWidth;
        this.height = window.innerHeight;
        window.addEventListener('resize', () => this.update());
    }

    update() {
        this.width = window.innerWidth;
        this.height = window.innerHeight;
    }

    getWidth() { return this.width; }
    getHeight() { return this.height; }

    isMobile() {
        return this.width <= 768 || ('ontouchstart' in window);
    }

    getGameAreaSize() {
        const maxW = Math.min(this.width, 900);
        const maxH = Math.min(this.height, 700);
        return { width: maxW, height: maxH };
    }

    getCellSize(cols = 30, rows = 20) {
        const area = this.getGameAreaSize();
        return Math.floor(Math.min(area.width / cols, area.height / rows));
    }
}
